File: dotnet-backend/Emart.Api/DTOs/ApiResponse.cs
using System;

namespace Emart.Api.DTOs
{
    public class ApiResponse<T>
    {
        public bool Success { get; set; }
        public string Message { get; set; } = string.Empty;
        public T? Data { get; set; }
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;

        public ApiResponse() { }

        public ApiResponse(bool success, string message, T? data)
        {
            Success = success;
            Message = message;
            Data = data;
        }

        public static ApiResponse<T> CreateSuccess(string message, T data)
        {
            return new ApiResponse<T>(true, message, data);
        }

        public static ApiResponse<T> CreateSuccess(string message)
        {
            return new ApiResponse<T>(true, message, default);
        }
    }
}

File: backend-email-microservice/src/main/java/com/example/demo/response/ApiResponse.java
package com.example.demo.response;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.LocalDateTime;

/**
 * The same success envelope the two backends already return, repeated here so
 * a caller parses one shape no matter which of the three services it talked to:
 *
 *   { "success": true, "message": "...", "data": { ... }, "timestamp": "..." }
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ApiResponse<T>(boolean success, String message, T data, LocalDateTime timestamp) {

    public static <T> ApiResponse<T> success(String message, T data) {
        return new ApiResponse<>(true, message, data, LocalDateTime.now());
    }

    public static <T> ApiResponse<T> success(String message) {
        return new ApiResponse<>(true, message, null, LocalDateTime.now());
    }
}
